lms/views/predicates/_lti_launch.py

```diff
@@ -162,7 +162,7 @@ def authorized_assignment_launch_view(context, request):
     name = "authorized_to_configure_assignments"
 
     def __call__(self, context, request):
-        roles = request.params.get("roles", "").lower()
+        roles = request.lti_user.roles.lower()
```

Comment:
So this line will now work for authorization param-authenticated requests (where there's no `request.params["roles"]`) as well as for actual launch requests

Reply:
P.S. Yes this "Is the user an administrator, instructor or teachingassistant?" logic is duplicated, that's a separate problem